sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -899,6 +918,22 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
           spec, partitionColumnNames, tablePath)
         try {
           tablePath.getFileSystem(hadoopConf).rename(wrongPath, rightPath)
+
+          // If the newSpec contains more than one depth partition, FileSystem.rename just deletes
+          // the leaf(i.e. wrongPath), we should check if wrongPath's parents need to be deleted.
+          // for example:
+          // newSpec is 'A=1/B=2', after renamePartitions by Hive, the location path in FileSystem
+          // is changed to 'a=1/b=2', which is wrongPath, then we renamed to 'A=1/B=2', and
+          // 'a=1/b=2' in FileSystem is deleted, while 'a=1' is already exists,
+          // which should also be deleted
+          val delHivePartPathAfterRename = getExtraPartPathCreatedByHive(
+            lowerCasePartitionSpec(spec),
```

Comment:
The last comment: I prefer to calling `lowerCasePartitionSpec ` in the func `getExtraPartPathCreatedByHive`.